### PR TITLE
Add regression test for #8395

### PR DESCRIPTION
**Pull request by ddeville**
## What do these changes do?

Introduce a regression test to help fixing #8395.

## Are there changes in behavior for the user?

No this is just a test.

## Is it a substantial burden for the maintainers to support this?

Shouldn't be :) The idea here is to help with catching regressions like these in the future.

## Related issue number

#8395

## Checklist

- [x] I think the code is well written
- [x] Unit tests for the changes exist
- [ ] Documentation reflects the changes
- [ ] If you provide code modification, please add yourself to `CONTRIBUTORS.txt`
  * The format is &lt;Name&gt; &lt;Surname&gt;.
  * Please keep alphabetical order, the file is sorted by names.
- [ ] Add a new news fragment into the `CHANGES/` folder
  * name it `<issue_or_pr_num>.<type>.rst` (e.g. `588.bugfix.rst`)
  * if you don't have an issue number, change it to the pull request
    number after creating the PR
    * `.bugfix`: A bug fix for something the maintainers deemed an
      improper undesired behavior that got corrected to match
      pre-agreed expectations.
    * `.feature`: A new behavior, public APIs. That sort of stuff.
    * `.deprecation`: A declaration of future API removals and breaking
      changes in behavior.
    * `.breaking`: When something public is removed in a breaking way.
      Could be deprecated in an earlier release.
    * `.doc`: Notable updates to the documentation structure or build
      process.
    * `.packaging`: Notes for downstreams about unobvious side effects
      and tooling. Changes in the test invocation considerations and
      runtime assumptions.
    * `.contrib`: Stuff that affects the contributor experience. e.g.
      Running tests, building the docs, setting up the development
      environment.
    * `.misc`: Changes that are hard to assign to any of the above
      categories.
  * Make sure to use full sentences with correct case and punctuation,
    for example:
    ```rst
    Fixed issue with non-ascii contents in doctest text files
    -- by :user:`contributor-gh-handle`.
    ```

    Use the past tense or the present tense a non-imperative mood,
    referring to what's changed compared to the last released version
    of this project.

### CODE REVIEW

File: tests/test_client_functional.py [hunk 1]
```diff
@@ -3644,3 +3644,26 @@ async def handler(_: web.Request) -> web.Response:
...
+    app.add_routes([web.get("/", handler)])
+    client = await aiohttp_client(app)
+
+    try:
+        await client.get("/")
+    except aiohttp.ClientResponseError as e:
+        assert e.status == 400
+        assert "Got more than 8190 bytes" in e.message
+    else:
+        raise AssertionError("Expected ClientResponseError")
```
Comment (Dreamsorcerer) [inline tests/test_client_functional.py]: I think the conventional way to do this is using pytest.raises().

Comment (webknjaz) [reply tests/test_client_functional.py]: Yep, `with pytest.raises()` is the approach to be used with pytest. Also, don't forget to set that `match=` regex in the invocation.

Comment (webknjaz) [reply tests/test_client_functional.py]: @ddeville could you address this? Also, if you add the test in a separate commit, then wait for like 30 minutes and push another commit with just the `xfail` marker, we'll have the logs for the failure in our CI which may be useful.

Comment (Dreamsorcerer) [reply tests/test_client_functional.py]: > Also, if you add the test in a separate commit, then wait for like 30 minutes and push another commit with just the `xfail` marker, we'll have the logs for the failure in our CI which may be useful.

Not necessary, xfails are in the logs:
https://github.com/aio-libs/aiohttp/actions/runs/8914204390/job/24508183641?pr=8396#step:11:4697

Comment (ddeville) [reply tests/test_client_functional.py]: Just updated to use `pytest.raises`.

Comment (webknjaz) [reply tests/test_client_functional.py]: > Not necessary, xfails are in the logs: https://github.com/aio-libs/aiohttp/actions/runs/8914204390/job/24508183641?pr=8396#step:11:4697

Ah, I didn't realize. This must be something new. I don't think the older pytest versions printed out such details for xfails...

**Review by webknjaz — APPROVED** [review]
Thanks!

File: tests/test_client_functional.py (hunk 1 above)
Comment (webknjaz) [reply tests/test_client_functional.py]: Marking this thread as resolved.

**Pull request merged**